cabal-install/tests/UnitTests/Distribution/Client/Init/FileCreators.hs

```diff
@@ -0,0 +1,171 @@
+module UnitTests.Distribution.Client.Init.FileCreators (
+  tests
+  ) where
+
+import Distribution.Client.Init.FileCreators
+  ( generateCabalFile )
+
+import qualified Data.Set as Set
+
+import Distribution.Client.Init.Types
+  ( InitFlags(..), PackageType(..) )
+import Distribution.Simple.Setup
+  ( Flag(..) )
+
+
+import Distribution.CabalSpecVersion
+  ( CabalSpecVersion(CabalSpecV2_4) )
+import Distribution.Types.Dependency
+  ( Dependency, mkDependency )
+import Distribution.Types.PackageName
+  ( mkPackageName )
+import Distribution.Types.VersionRange
+  ( majorBoundVersion )
+import Distribution.Types.Version
+  ( mkVersion )
+import qualified Distribution.ModuleName as ModuleName
+  ( fromString )
+import qualified Distribution.SPDX as SPDX
+import Language.Haskell.Extension ( Language(..) )
+
+import Test.Tasty
+import Test.Tasty.HUnit
+
+tests :: [TestTree]
+tests =
+  [ testCase "generateCabalFile exe"
+    (generateCabalFileTest "exekg" exeFlags expectedExeOutput)
+  , testCase "generateCabalFile libExeAndTests"
+    (generateCabalFileTest "libExeAndTestsPackage" libExeAndTestFlags expectedLibExeAndTestOutput)
+  ]
+
+generateCabalFileTest :: String -> InitFlags -> String -> Assertion
+generateCabalFileTest filename flags expected =
+  assertEqual
+  ("Generated cabal file does not match: '" ++ filename ++ "'") expected actual
+  where
+    actual = generateCabalFile filename flags
+
+--------------------------------------------------
+-- Simple exe.
+
+exeFlags :: InitFlags
+exeFlags = baseTestFlags {
+    packageType = Flag Executable
+  , applicationDirs = Just ["app"]
+  }
+
+expectedExeOutput :: String
+expectedExeOutput =
+     "cabal-version:       2.4\n"
```

Comment:
Use https://hackage.haskell.org/package/tasty-golden

Updating these by hand will be PITA. `tasty-golden` allows run tests with `--accept`. Cabal have some test using golden-tests already (roundtrip in `parser-tests`)

Reply:
TIL, thanks! That's so much better.